java101/src/main/java/com/java101/HeroController.java

```diff
@@ -3,20 +3,55 @@
 import java.io.IOException;
 import java.util.List;
 
+import javax.annotation.PostConstruct;
+
 import org.json.simple.parser.ParseException;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
 
 @Controller
 public class HeroController {
-
-	@RequestMapping("/heroes")
+	
+	private List<Hero> heroes;
+	
+	@Autowired
+	private HeroService heroService;
+	
+	@PostConstruct
+	public void init() throws IOException, ParseException {
+		heroes = heroService.createHeroesListFromJsonFile();
+	}
+	
+	@RequestMapping(value = "/heroes", method = RequestMethod.GET)
 	public String heroes(Model model) throws IOException, ParseException {
-		HeroService heroService = new HeroService();
-		List<Hero> heroes = heroService.createHeroesListFromJsonFile();
 		model.addAttribute("heroes", heroes);
 		return "heroes";
 	}
+	
+	@RequestMapping(value = "/addHero", method = RequestMethod.GET)
+	public String addHeroForm(Model model) {
+		return "addHero";
+	}
+	
+    @RequestMapping(value = "/addHero", method = RequestMethod.POST)
+    public String add(@ModelAttribute("hero") Hero hero) {
+
+        if (null != hero 
+        	&& null != hero.getName()
+            && 0 != hero.getPower() 
+            && 0 != hero.getSkill() 
+            && 0 != hero.getVitality() 
+            && !hero.getName().isEmpty()) {
```

Comment:
O walidacji danych będzie osobne zadanie, także na razie może być, ale wynieś to chociaż do osobnej metody. Możesz tą metodę nawet w klasie Hero umieścić